rotkehlchen/tests/data_migrations/test_migrations.py

```diff
@@ -285,3 +285,97 @@ def test_migration_5(rotkehlchen_api_server):
+
+
+@pytest.mark.parametrize('data_migration_version', [None])
+@pytest.mark.parametrize('perform_migrations_at_unlock', [False])
+@pytest.mark.parametrize('perform_upgrades_at_unlock', [False])
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+@pytest.mark.parametrize('perform_nodes_insertion', [False])
+def test_migration_6_default_nodes(rotkehlchen_api_server):
+    """
+    Test that the sixs data migration works when the user has not customized the nodes.
+    """
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    database = rotki.data.db
+    dir_path = Path(__file__).resolve().parent.parent.parent
+    with open(dir_path / 'data' / 'nodes_as_of_1-26-1.json', 'r') as f:
+        old_nodes_info = json.loads(f.read())
+    with open(dir_path / 'data' / 'nodes.json', 'r') as f:
+        new_nodes_info = json.loads(f.read())
+
+    old_nodes = [
+        (node['name'], node['endpoint'], node['owned'], node['weight'], node['active'], node['blockchain'])  # noqa: E501
+        for node in old_nodes_info
+    ]
+
+    with database.user_write() as write_cursor:
+        write_cursor.executemany(
+            'INSERT INTO web3_nodes (name, endpoint, owned, weight, active, blockchain) VALUES (?, ?, ?, ?, ?, ?)',  # noqa: E501
+            old_nodes,
+        )
+
+    migration_patch = patch(
+        'rotkehlchen.data_migrations.manager.MIGRATION_LIST',
+        new=[MIGRATION_LIST[5]],
+    )
+    with migration_patch:
+        DataMigrationManager(rotki).maybe_migrate_data()
+
+    # Now nodes should be the new defaults
+    with database.conn.read_ctx() as cursor:
+        nodes_in_db = set(cursor.execute('SELECT name, endpoint, owned, weight, blockchain FROM web3_nodes'))  # noqa: E501
+
+    new_nodes = {
+        (node['name'], node['endpoint'], node['owned'], node['weight'], node['blockchain'])  # noqa: E501
+        for node in new_nodes_info
+    }
+
+    assert nodes_in_db == new_nodes
+
+
+@pytest.mark.parametrize('data_migration_version', [None])
+@pytest.mark.parametrize('perform_migrations_at_unlock', [False])
+@pytest.mark.parametrize('perform_upgrades_at_unlock', [False])
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+@pytest.mark.parametrize('perform_nodes_insertion', [False])
+def test_migration_6_customized_nodes(rotkehlchen_api_server):
+    """
+    Test that the sixs data migration works when the user has some cusotmized the nodes.
+    """
+    rotki = rotkehlchen_api_server.rest_api.rotkehlchen
+    database = rotki.data.db
+    dir_path = Path(__file__).resolve().parent.parent.parent
+    with open(dir_path / 'data' / 'nodes_as_of_1-26-1.json', 'r') as f:
+        old_nodes_info = json.loads(f.read())
+
+    old_nodes = [
+        [node['name'], node['endpoint'], node['owned'], node['weight'], node['active'], node['blockchain']]  # noqa: E501
+        for node in old_nodes_info
+    ]
+    old_nodes[0][0] = 'Renamed etherscan'  # customize a node
```

Comment:
This node can't be customixed irc. Can you add a new one in the database, for example from the list of new ones and see that it doesn't get deleted?

Reply:
Adding another node is a bit more tricky since in requires to reweigh the nodes. I changed this line so that now the cloudflare node is being renamed. Is it okay in your opinion?

Reply:
For the intention of testing is required to rebalance? The weights adding one is needed when we call random from python as is a constraint in the library. There is no other place in the logic where that constraint is needed afaik. Also is just calling one extra function if you do it.

I would suggest to add a new node since it will test a more realistic scenario of users having new nodes vs just editing the name (I would do both, editing name and new node)

Reply:
Ok, added also a new node